core/src/main/java/org/springframework/security/authentication/ProviderManager.java

```diff
@@ -145,7 +145,7 @@ private void checkState() {
 			throw new IllegalArgumentException(
 					"A parent AuthenticationManager or a list "
 							+ "of AuthenticationProviders is required");
-		} else if (providers.contains(null)) {
+		} else if (providers.stream().anyMatch(Objects::isNull)) {
```

Comment:
At the framework level, we find that the Stream API is not fast enough. Would you please use a for loop instead?

Reply:
In particular we avoid the Stream APIs because they create extra objects which leads to excessive Garbage Collection. This can cause the JDK to pause and deteriorate the performance. See gh-7154

I believe we could use Spring's [CollectionUtils.contains](https://github.com/spring-projects/spring-framework/blob/master/spring-core/src/main/java/org/springframework/util/CollectionUtils.java#L125)

Reply:
I've understood it and amended to use `CollectionUtils.contains`.